react-native/android/build.gradle

```diff
@@ -31,10 +31,17 @@ import org.apache.tools.ant.filters.ReplaceTokens
 // We download various C++ open-source dependencies into downloads.
 // We then copy both the downloaded code and our custom makefiles and headers into third-party-ndk.
 // After that we build native code from src/main/jni with module path pointing at third-party-ndk.
+def npmExecutable = 'npm';
 
-ext.coreVersion = "npm --silent run get-core-version".execute().text.trim()
-ext.syncVersion = "npm --silent run get-sync-version".execute().text.trim()
-def currentVersion = "npm --silent run get-version".execute().text.trim()
+if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        npmExecutable = 'npm.cmd'
+}
+ext.coreVersion = "$npmExecutable --silent run get-core-version".execute().text.trim()
```

Comment:
I tried to reduce the churn here and preserve the logic as much as possible. These lines were failing on windows hence this fix. Also we are using the values from the dependencies.list as environment variables on some places during the build so it will require us to do the same thing that the npm helper package does for us.